x-pack/legacy/plugins/ml/server/routes/results_service.ts

```diff
@@ -0,0 +1,220 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { RequestHandlerContext } from 'src/core/server';
+import { schema } from '@kbn/config-schema';
+import { licensePreRoutingFactory } from '../new_platform/licence_check_pre_routing_factory';
+import { wrapError } from '../client/error_wrapper';
+import { RouteInitialization } from '../new_platform/plugin';
+import {
+  anomaliesTableDataSchema,
+  partitionFieldValuesSchema,
+} from '../new_platform/results_service_schema';
+import { resultsServiceProvider } from '../models/results_service';
+
+function getAnomaliesTableData(context: RequestHandlerContext, payload: any) {
+  const rs = resultsServiceProvider(context);
+  const {
+    jobIds,
+    criteriaFields,
+    influencers,
+    aggregationInterval,
+    threshold,
+    earliestMs,
+    latestMs,
+    dateFormatTz,
+    maxRecords,
+    maxExamples,
+    influencersFilterQuery,
+  } = payload;
+  return rs.getAnomaliesTableData(
+    jobIds,
+    criteriaFields,
+    influencers,
+    aggregationInterval,
+    threshold,
+    earliestMs,
+    latestMs,
+    dateFormatTz,
+    maxRecords,
+    maxExamples,
+    influencersFilterQuery
+  );
+}
+
+function getCategoryDefinition(context: RequestHandlerContext, payload: any) {
+  const rs = resultsServiceProvider(context);
+  return rs.getCategoryDefinition(payload.jobId, payload.categoryId);
+}
+
+function getCategoryExamples(context: RequestHandlerContext, payload: any) {
+  const rs = resultsServiceProvider(context);
+  const { jobId, categoryIds, maxExamples } = payload;
+  return rs.getCategoryExamples(jobId, categoryIds, maxExamples);
+}
+
+function getMaxAnomalyScore(context: RequestHandlerContext, payload: any) {
+  const rs = resultsServiceProvider(context);
+  const { jobIds, earliestMs, latestMs } = payload;
+  return rs.getMaxAnomalyScore(jobIds, earliestMs, latestMs);
+}
+
+function getPartitionFieldsValues(context: RequestHandlerContext, payload: any) {
+  const rs = resultsServiceProvider(context);
+  const { jobId, searchTerm, criteriaFields, earliestMs, latestMs } = payload;
+  return rs.getPartitionFieldsValues(jobId, searchTerm, criteriaFields, earliestMs, latestMs);
+}
+
+/**
+ * Routes for results service
+ */
+export function resultsServiceRoutes({ xpackMainPlugin, router }: RouteInitialization) {
+  /**
+   * @apiGroup ResultsService
+   *
+   * @api {post} /api/ml/results/anomalies_table_data Prepare anomalies records for table display
+   * @apiName GetAnomaliesTableData
+   * @apiDescription Retrieves anomaly records for an anomaly detection job and formats them for anomalies table display
+   */
+  router.post(
+    {
+      path: '/api/ml/results/anomalies_table_data',
+      validate: {
+        body: schema.object({ ...anomaliesTableDataSchema }),
```

Comment:
why do you need a shallow copy here?

Reply:
and all the other places with validate schemas

Reply:
I saw this pattern initially when I was looking for examples 🤔 Though you're right, we don't really need it. Happy to change these if it makes sense. 😄